packages/mdc-side-sheet/_mixins.scss

```diff
@@ -110,90 +113,127 @@
 // Private
 //
 
-@mixin mdc-side-sheet-stroke-right_($width) {
-  border-right-width: $width;
-  border-left-width: 0;
-  border-right-style: solid;
-  border-left-style: none;
-}
-
-@mixin mdc-side-sheet-stroke-left_($width) {
-  border-right-width: 0;
-  border-left-width: $width;
-  border-right-style: none;
-  border-left-style: solid;
-}
-
-@mixin mdc-side-sheet-stroke-width_($width, $side: trailing) {
-  @if $side == leading {
-    @include mdc-side-sheet-stroke-right_($width);
-
-    @include mdc-rtl {
-      @include mdc-side-sheet-stroke-left_($width);
-    }
-  } @else if $side == trailing {
-    @include mdc-side-sheet-stroke-left_($width);
-
-    @include mdc-rtl {
-      @include mdc-side-sheet-stroke-right_($width);
-    }
-  } @else {
-    @error "mdc-side-sheet-stroke_() mixin: $side must be 'leading' or 'trailing', but got '#{$side}'";
-  }
-}
-
 @mixin mdc-side-sheet-base_($side: trailing) {
+  @include mdc-typography-base;
+
+  $flexOrder: if($side == leading, 0, 1);
   $xFactor: if($side == leading, 1, -1);
 
-  order: if($side == leading, 0, 1);
+  display: flex;
+  flex-direction: column;
+  flex-shrink: 0;
+  order: $flexOrder;
+  box-sizing: border-box;
+  height: 100%;
+  transition-property: transform;
+  transition-timing-function: $mdc-animation-standard-curve-timing-function;
+  overflow: hidden;
+
+  #{&}--open {
+    transition: mdc-animation-standard(transform, $mdc-side-sheet-animation-enter);
+  }
 
-  &--animate {
+  #{&}--animate {
     transform: translateX(-100% * $xFactor);
 
     @include mdc-rtl {
       transform: translateX(100% * $xFactor);
     }
   }
 
-  &--opening {
+  #{&}--opening {
     transform: translateX(0);
+    transition-duration: $mdc-side-sheet-animation-enter;
 
     // Required to level the specificity with animate class.
     @include mdc-rtl {
       transform: translateX(0);
     }
   }
 
-  &--closing {
+  #{&}--closing {
     transform: translateX(-100% * $xFactor);
+    transition-duration: $mdc-side-sheet-animation-exit;
 
     @include mdc-rtl {
       transform: translateX(100% * $xFactor);
     }
   }
 }
 
-@mixin mdc-side-sheet-modal_($side: trailing) {
+@mixin mdc-side-sheet-dismissible_($rootClass, $side: trailing) {
```

Comment:
It's possible, yes, but it feels a little weird to me: 4d73a7a3e9834caedc85579e36b9f26699185f52

Specifically, `mdc-side-sheet.scss` ends up looking like this:

```scss
.mdc-side-sheet {
  @include mdc-side-sheet-base_;
  @include mdc-side-sheet-dismissible_;
  @include mdc-side-sheet-modal_;
  @include mdc-side-sheet-scrim_;
  @include mdc-side-sheet-stroke-width_(1px);
  @include mdc-side-sheet-stroke-color($mdc-side-sheet-stroke-color);
  @include mdc-side-sheet-surface-fill-color($mdc-side-sheet-surface-fill-color);
  // ...
}

.mdc-side-sheet--modal {
  @include mdc-side-sheet-scrim-fill-color($mdc-side-sheet-modal-scrim-color);
  @include mdc-elevation($mdc-side-sheet-modal-elevation);
}
```

1. The root CSS class calls mixins for itself _and_ the two variants _and_ the scrim, which feels odd.

2. The root CSS class calls a `modal_` mixin, but right below it, there's _also_ a `--modal` CSS class. I could move those two mixin calls into `mdc-side-sheet-modal_`, but we normally try to keep all the theme configuration in the main `mdc-foo.scss` file.

3. It forces most of the CSS classes to go in mixins instead of the main `mdc-side-sheet.scss` file, which IMO makes the styles harder to read and understand.
    - On a related note, `&--open` selectors make search/replace harder, and it's easy to forget about them when you're renaming something.

The `$rootClass` param isn't great, but it seems cleaner than the alternative. And the mixins that have it are all "private", so we don't need to worry about clients using it.

WDYT?

Reply:
My original concern was mostly just the required repetition of the base class which we could easily mess up in copypasta, but I guess that's a pretty small price to pay which should also be obvious in screenshot tests. So I'm willing to go with that since this didn't pan out.